### ISSUE TITLE
Are scope requirements meant to be transitive?

### ISSUE THREAD

**Issue by mikehearn** [avaje/avaje-inject#140]
I wrote the moral equivalent of this:

```

@Scope
@InjectModule(requires = [ A::class, B::class ])
annotation class Foo

@Scope
@InjectModule(requires = [ Foo::class, C::class ])
annotation class Bar

@Bar
class Baz(val a: A, val b: B)
```

I was expecting that because scope Bar depends on scope Foo, and Foo depends on A (i.e. it's externally provided) that Baz would be able to depend on A. However it seems that doesn't work. I get errors about missing dependencies. Is that intentional? If so, is the right thing to do just duplicating the requires=[] list ?

**Comment by rbygrave** [comment]
> Baz would be able to depend on A

Yes, that should work. Baz should be able to depend on A.

> However it seems that doesn't work. I get errors about missing dependencies. 

Should be a bug. 

We have `ParentScopeTest` but that doesn't test external dependencies on the parent scope which is the case here with the Foo scope above.  I'll look to create a test case where the parent scope has external dependencies and get that to reproduce.

**Comment by rbygrave** [comment]
Ok, fixed in 6.6

**Comment by mikehearn** [comment]
Awesome, thanks.
